packages/slice-machine/lib/mock/CustomType.ts

```diff
@@ -1,88 +1,54 @@
+  const widgets = flattenWidgets(model);
+  const widgetsConfig = widgets.reduce<
+    Partial<Record<WidgetKey, DocWidgetMockConfig>>
+  >((acc, [key, w]) => {
+    const legacyFieldConfig: Partial<Record<string, unknown>> | undefined =
+      legacyMockConfig[key];
+    if (!legacyFieldConfig) return acc;
+
+    const widgetConfig = buildWidgetMockConfig(w, legacyFieldConfig);
+    if (!widgetConfig) return acc;
+
+    return { ...acc, [key]: widgetConfig };
+  }, {});
+
+  return { value: widgetsConfig };
 }
 
-const fieldsHandler = handleFields(Widgets);
-
-const groupHandler = (group: GroupSM, mockConfig: CustomTypeMockConfig) => {
-  const items = [];
-  const entries = group.config?.fields;
-  for (let i = 0; i < Math.floor(Math.random() * 6) + 2; i++) {
-    items.push(fieldsHandler(entries, mockConfig));
-  }
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  return items;
-};
-
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const sliceZoneHandler = () => {};
-
-const createEmptyMock = (type: string) => ({
-  id: faker.datatype.uuid(),
-  uid: null,
-  type,
-  data: {},
-});
-
-// eslint-disable-next-line @typescript-eslint/require-await
-export default async function MockCustomType(
+export default function MockCustomType(
   model: CustomTypeSM,
-  mockConfig: CustomTypeMockConfig
+  legacyMockConfig: CustomTypeMockConfig
 ) {
-  const customTypeMock: Mock = createEmptyMock(model.id);
-  const maybeUid = model.tabs.reduce((acc, curr) => {
-    const maybeWidgetUid = Object.entries(curr.value).find(
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-unsafe-member-access
-      ([_, e]) => e.value.type === "UID"
-    );
-    if (!acc && maybeWidgetUid) {
-      return curr;
-    }
-    return acc;
-  });
-
-  if (maybeUid) {
-    customTypeMock.uid = Widgets.UID.handleMockConfig();
-  }
-
-  for (const [, tab] of Object.entries(model.tabs)) {
-    const { fields, groups, sliceZone } = Tab.organiseFields(tab);
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const mockedFields = fieldsHandler(fields as [], mockConfig);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    customTypeMock.data = {
-      ...customTypeMock.data,
-      ...mockedFields,
-    };
-    groups.forEach((group) => {
-      const { key, value } = group;
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
-      const groupMockConfig = CustomTypeMockConfig.getFieldMockConfig(
-        mockConfig,
-        key
-      );
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      const groupFields = groupHandler(value, groupMockConfig);
-      customTypeMock.data[key] = groupFields;
-    });
-
-    if (sliceZone) {
-      sliceZoneHandler();
-    }
-  }
-  return customTypeMock;
+  const prismicModel = CustomTypes.fromSM(model);
```

Comment:
Not at all, CustomTypeSM a used everywhere in SM, it should be converted back to the prismic model only if writing it on FS and sending it to the custom types API that's it, the server has SM types everywhere too.
Here we convert it simply because the mocks lib expects a prismic model.

Reply:
I do understand that it was just for this case about using external libs expecting this type.
I just think those cases might happen often (pushing data, writing other files, ...).

So i'm wondering if this is not better to have the server handle data compatible with other APIs by default and only use the SM models for rendering purpose (which is why they were created).

Reply:
I don't think that client/server is good criteria to decide where we should use a given model.
Slicemachine UI is meant to have its own model with its own constraints and everything so I'd rather keep it that way and only jump back to prismic model when contacting prismic external apps, it's more straightforward and meaningful in that case. A model should correspond to a usage, not based on where it's used

Reply:
Alright, I agree then that the whole UI as a specific model.
In that optic I suggest that all the functions related to transforming a model into the other one and vice-versa should be located in the IO module.

Reply:
I disagree on the location, it's still model related to go back and forth with prismic, IO is only about input output so writing and reading the filesystem in that case. And that conversion is specific to SM hence the `model` folder of SM close to the SM model.